packages/ui-components/src/FormComponents/MetadataEditorTags.tsx

```diff
@@ -137,130 +106,121 @@ export class MetadataEditorTags extends React.Component<
-                        id={'editor' + '-' + tag + '-' + index}
-                        key={'editor' + '-' + tag + '-' + index}
-                      >
-                        {tag}
-                        <checkIcon.react
-                          tag="span"
-                          elementPosition="center"
-                          height="18px"
-                          width="18px"
-                          marginLeft="5px"
-                          marginRight="-3px"
-                        />
-                      </button>
-                    );
-                  } else {
-                    return (
-                      <button
-                        onClick={this.handleClick}
-                        className={`${METADATA_EDITOR_TAG} tag unapplied-tag`}
-                        id={'editor' + '-' + tag + '-' + index}
-                        key={'editor' + '-' + tag + '-' + index}
-                      >
-                        {tag}
-                      </button>
-                    );
-                  }
-                })()
-              )
-            : null}
-          {inputBox}
-        </li>
-      </div>
+    setAddingNewTag(false);
+  };
+
+  const hasTags = tags;
+  const inputBox =
+    addingNewTag === true ? (
+      <ul
+        className={`${METADATA_EDITOR_TAG} tag unapplied-tag`}
+        key={'editor-new-tag'}
+      >
+        <input
+          className={`${METADATA_EDITOR_INPUT_TAG}`}
+          onClick={(
+            event: React.MouseEvent<HTMLInputElement, MouseEvent>
+          ): void => addTagOnClick(event)}
+          onKeyDown={async (
+            event: React.KeyboardEvent<HTMLInputElement>
+          ): Promise<void> => {
+            await addTagOnKeyDown(event);
+          }}
+          onBlur={(event: React.FocusEvent<HTMLInputElement>): void =>
+            addTagOnBlur(event)
+          }
+          autoFocus
+        />
+      </ul>
+    ) : (
+      <button
+        onClick={(): void => setAddingNewTag(true)}
+        className={`${METADATA_EDITOR_TAG} tag unapplied-tag`}
+      >
+        Add Tag
+        <addIcon.react
+          tag="span"
+          className={METADATA_EDITOR_TAG_PLUS_ICON}
+          elementPosition="center"
+          height="16px"
+          width="16px"
+          marginLeft="2px"
+        />
+      </button>
     );
-  }
 
-  render(): JSX.Element {
-    return <div>{this.renderTags()}</div>;
-  }
-}
+  return (
+    <div>
+      <label className="control-label"> Tags </label>
+      <li className={METADATA_EDITOR_TAG_LIST}>
+        {hasTags
+          ? tags.map((tag: string, index: number) =>
+              ((): JSX.Element => {
+                if (!selectedTags) {
+                  return (
+                    <button
+                      onClick={handleClick}
+                      className={`${METADATA_EDITOR_TAG} tag unapplied-tag`}
+                      id={'editor' + '-' + tag + '-' + index}
```

Comment:
why not use formatted strings for the strings in the return block (instead of using +)